docs/coreneuron/getting-coreneuron.rst

```diff
@@ -0,0 +1,167 @@
+
+Step-by-step instructions
+=========================
+As stated above, the simplest way to enable CoreNEURON is to build it as a git submodule of the NEURON repository.
+This section outlines how to do that on a typical system, using the ``master`` version of NEURON.
+
+First, we clone the latest version of NEURON:
+
+.. code-block::
+
+   git clone https://github.com/neuronsimulator/nrn
+   cd nrn
+
+Then we create a build directory:
+
+.. code-block::
+
+   mkdir build
+   cd build
+
+In order to configure NEURON, we need to load the required software dependencies.
+If compilers and necessary dependencies are already available in the default paths then you do not need to do anything.
+In a cluster or HPC environment a module system is often used to select software.
+For example, you might be able to load the compiler, cmake, and python dependencies using ``module``:
+
+.. code-block::
+
+   module load intel openmpi python cmake
+
+And to enable GPU support then you might additionally load NVIDIA HPC SDK and CUDA modules:
+
+.. code-block::
+
+   module load nvidia-hpc-sdk cuda
+
+.. warning::
+   Module names are not standardised across different systems, and you will almost certainly need to adjust these names for your system
+   ``module avail`` and ``module spider`` may help you to discover the module names that you need to use.
+
+
+.. warning::
+   The NVIDIA HPC SDK bundles up to four different versions of CUDA.
+   You should make sure that the version of CUDA in your environment is compatible with the CUDA driver on your system, otherwise you may encounter errors like
+   ``nvlink fatal: Input file '[snip]' newer than toolkit (112 vs 110) (target: sm_60)``
+
+Once the required dependencies are loaded, you are ready to run CMake. See :ref:`Install NEURON using CMake` for more information.
+To enable CoreNEURON, don't forget to add the ``-DNRN_ENABLE_CORENEURON=ON`` option.
+
+.. code-block::
+
+   cmake .. \
+     -DNRN_ENABLE_CORENEURON=ON \
+     -DNRN_ENABLE_INTERVIEWS=OFF \
+     -DNRN_ENABLE_RX3D=OFF \
+     -DCMAKE_INSTALL_PREFIX=$HOME/install \
+     -DCMAKE_C_COMPILER=icc \
+     -DCMAKE_CXX_COMPILER=icpc
+
+Make sure to replace ``icc`` and ``icpc`` with the C/C++ compiler that you are using.
+Also change `$HOME/install` to desired installation directory.
+CMake tries to find MPI libraries automatically but if needed you can set MPI compiler options ``-DMPI_C_COMPILER=<mpi C compiler>`` and ``-DMPI_CXX_COMPILER=<mpi CXX compiler>``.
+
+If you would like to enable GPU support with OpenACC, make sure to use ``-DCORENRN_ENABLE_GPU=ON`` option and to use the PGI/NVIDIA HPC SDK compilers with CUDA.
+For example,
+
+.. code-block::
+
+   cmake .. \
+     -DNRN_ENABLE_CORENEURON=ON \
+     -DCORENRN_ENABLE_GPU=ON \
+     -DNRN_ENABLE_INTERVIEWS=OFF \
+     -DNRN_ENABLE_RX3D=OFF \
+     -DCMAKE_INSTALL_PREFIX=$HOME/install \
+     -DCMAKE_C_COMPILER=nvc \
+     -DCMAKE_CUDA_COMPILER=nvcc \
```

Comment:
I'm wondering if it's worth mentioning that `nvcc` could come either from the `cuda` module or the `nvhpc` toolkit installation